### DISCUSSION TITLE
"Predicting spectra and IMs" step lasts for more than 33 hours, and still running at this step.

### DISCUSSION THREAD

**Discussion by yangyangbian** [vdemichev/DiaNN#769] [Q&A]
Hi Dr Demichev,
I have run some dimethyl labeled peptides with DIA method.
Since some K in the protein might be labeled with CH2O.
In all the samples, the peptides might have either 14 Da or 28 Da mass change on K (The chance is about 50-60% of peptides labeled with dimethyl tag).
Therefore, I need to set variable modification of 14 and 28 Da in my parameter.

I set parameters like this:
Briefly, 1) I used library free search; 2) I used rat fasta file containing 84764 sequences; 3) I set some specific parameters in the additional panel. 4) I have tried this search before, and the "Predicting spectra and IMs" step is super long. I thought that mightbe something wrong, so I re-run it again.
**However,  the "Predicting spectra and IMs" step lasts for more than 33 hours, and still running at this step.**
Did I run something wrong？ Thanks a lot.

![image](https://github.com/vdemichev/DiaNN/assets/107282849/96321cdd-57d6-4b51-8c87-c48f2a1ed5e7)

![image](https://github.com/vdemichev/DiaNN/assets/107282849/c7a46fc9-3a27-42ee-a7cd-464daa6173aa)

![image](https://github.com/vdemichev/DiaNN/assets/107282849/2a9ee7fa-a47b-41de-9bfd-f800f9be2fac)

The search log is here:
diann.exe --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R1.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R1_rerun1.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R2.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R3.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R4.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R5.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_exp_R1.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_exp_R2.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_exp_R3.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_exp_R4.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_exp_R5.raw
" --lib "" --threads 48 --verbose 1 --out "H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\report.tsv" --qvalue 0.01 --matrices  --out-lib "H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\output_5th_report.tsv" --gen-spec-lib --predictor --prosit --fasta "H:\0_fasta\UniProtKB_Rat_Rev_Unrev_54764_seq_20230306.fasta" --fasta-search --min-fr-mz 200 --max-fr-mz 1800 --met-excision --cut K*,R* --missed-cleavages 1 --min-pep-len 7 --max-pep-len 30 --min-pr-mz 360 --max-pr-mz 1500 --min-pr-charge 1 --max-pr-charge 4 --unimod4 --var-mods 1 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,*n --monitor-mod UniMod:1 --reanalyse --relaxed-prot-inf --smart-profiling --peak-center --no-ifs-removal --var-mod UniMod:1001,16.0282,K
--var-mod UniMod:1002,32.0564,K
--var-mods 4
--strip-unknown-mods
--missed-cleavages 3
--no-cut-after-mod [1001]
--no-cut-after-mod [1002]
DIA-NN 1.8.1 (Data-Independent Acquisition by Neural Networks)
Compiled on Apr 14 2022 15:31:19
Current date and time: Fri Jul 28 06:39:58 2023
CPU: GenuineIntel Intel(R) Xeon(R) Platinum 8331A CPU @ 2.50GHz
SIMD instructions: AVX AVX2 AVX512CD AVX512F FMA SSE4.1 SSE4.2 
Logical CPU cores: 48
Thread number set to 48
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
A spectral library will be generated
Deep learning will be used to generate a new in silico spectral library from peptides provided
Library-free search enabled
Min fragment m/z set to 200
Max fragment m/z set to 1800
N-terminal methionine excision enabled
In silico digest will involve cuts at K*,R*
Maximum number of missed cleavages set to 1
Min peptide length set to 7
Max peptide length set to 30
Min precursor m/z set to 360
Max precursor m/z set to 1500
Min precursor charge set to 1
Max precursor charge set to 4
Cysteine carbamidomethylation enabled as a fixed modification
Maximum number of variable modifications set to 1
Modification UniMod:35 with mass delta 15.9949 at M will be considered as variable
Modification UniMod:1 with mass delta 42.0106 at *n will be considered as variable
A spectral library will be created from the DIA runs and used to reanalyse them; .quant files will only be saved to disk during the first step
Highly heuristic protein grouping will be used, to reduce the number of protein groups obtained; this mode is recommended for benchmarking protein ID numbers; use with caution for anything else
When generating a spectral library, in silico predicted spectra will be retained if deemed more reliable than experimental ones
Fixed-width center of each elution peak will be used for quantification
Interference removal from fragment elution curves disabled
Modification UniMod:1001 with mass delta 16.0282 at K will be considered as variable
Modification UniMod:1002 with mass delta 32.0564 at K will be considered as variable
Maximum number of variable modifications set to 4
DIA-NN will use deep learning to predict spectra/RTs/IMs even for peptides carrying modifications which are not recognised by the deep learning predictor. In this scenario, if also generating a spectral library from the DIA data or using the MBR mode, it might or might not be better (depends on the data) to also use the --out-measured-rt option - it's recommended to test it with and without this option
Maximum number of missed cleavages set to 3
DIA-NN will optimise the mass accuracy automatically using the first run in the experiment. This is useful primarily for quick initial analyses, when it is not yet known which mass accuracy setting works best for a particular acquisition scheme.
Exclusion of fragments shared between heavy and light peptides from quantification is not supported in FASTA digest mode - disabled; to enable, generate an in silico predicted spectral library and analyse with this library
The following variable modifications will be scored: UniMod:1 

11 files will be processed
[0:00] Loading FASTA H:\0_fasta\UniProtKB_Rat_Rev_Unrev_54764_seq_20230306.fasta
[1:45] Processing FASTA
[7:29] Assembling elution groups
[19:47] 108544398 precursors generated
[21:43] Prosit input saved to H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\output_5th_report.prosit.csv
[21:49] Gene names missing for some isoforms
[21:49] Library contains 54690 proteins, and 24377 genes
[22:36] Encoding peptides for spectra and RTs prediction
[29:56] Predicting spectra and IMs

**Comment by yangyangbian** [comment]
I waited for more than 2 days, the step is finished.
However, the search stopped again.
Here is all the log information.

11 files will be processed
[0:00] Loading FASTA H:\0_fasta\UniProtKB_Rat_Rev_Unrev_54764_seq_20230306.fasta
[1:45] Processing FASTA
[7:29] Assembling elution groups
[19:47] 108544398 precursors generated
[21:43] Prosit input saved to H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\output_5th_report.prosit.csv
[21:49] Gene names missing for some isoforms
[21:49] Library contains 54690 proteins, and 24377 genes
[22:36] Encoding peptides for spectra and RTs prediction
[29:56] Predicting spectra and IMs
[3154:31] Predicting RTs
[3812:40] Decoding predicted spectra and IMs
[3821:44] Decoding RTs
[3832:08] Saving the library to H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\output_5th_report.predicted.speclib
[3841:22] Initialising library

[3842:35] First pass: generating a spectral library from DIA data
[3842:35] File #1/11
[3842:36] Loading run H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R1.raw
[3852:11] 108452019 library precursors are potentially detectable
[3852:20] Processing...
[3885:14] RT window set to 8.42353
[3885:14] Peak width: 3.824
[3885:14] Scan window radius set to 8
[3885:25] Recommended MS1 mass accuracy setting: 21.3388 ppm
[3928:44] Optimised mass accuracy: 17.4178 ppm


DIA-NN exited
DIA-NN-plotter.exe "H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\report.stats.tsv" "H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\report.tsv" "H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\report.pdf"
PDF report will be generated in the background

**Comment by yangyangbian** [comment]
I re-run the DIA-NN search, it stopped again with the 1st raw data.
stopped at "[145:51] Training neural networks: 85438 targets, 92857 decoys"



Here is all the log information.
diann.exe --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R1.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R1_rerun1.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R2.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R3.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R4.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R5.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_exp_R1.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_exp_R2.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_exp_R3.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_exp_R4.raw
" --f "H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_exp_R5.raw
" --lib "H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\2nd_test_no-cut-after_mod_K\output_5th_report.predicted.speclib" --threads 48 --verbose 1 --out "H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\report.tsv" --qvalue 0.01 --matrices  --out-lib "H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\output_5th_report.tsv" --gen-spec-lib --prosit --fasta "H:\0_fasta\UniProtKB_Rat_Rev_Unrev_54764_seq_20230306.fasta" --met-excision --cut K*,R* --missed-cleavages 1 --min-pep-len 7 --max-pep-len 30 --min-pr-mz 360 --max-pr-mz 1500 --min-pr-charge 1 --max-pr-charge 4 --unimod4 --var-mods 1 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,*n --monitor-mod UniMod:1 --reanalyse --relaxed-prot-inf --smart-profiling --peak-center --no-ifs-removal --var-mod UniMod:1001,16.0282,K
--var-mod UniMod:1002,32.0564,K
--var-mods 4
--strip-unknown-mods
--missed-cleavages 3
--no-cut-after-mod [1001]
--no-cut-after-mod [1002]
DIA-NN 1.8.1 (Data-Independent Acquisition by Neural Networks)
Compiled on Apr 14 2022 15:31:19
Current date and time: Mon Jul 31 11:04:13 2023
CPU: GenuineIntel Intel(R) Xeon(R) Platinum 8331A CPU @ 2.50GHz
SIMD instructions: AVX AVX2 AVX512CD AVX512F FMA SSE4.1 SSE4.2 
Logical CPU cores: 48
Thread number set to 48
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
A spectral library will be generated
N-terminal methionine excision enabled
In silico digest will involve cuts at K*,R*
Maximum number of missed cleavages set to 1
Min peptide length set to 7
Max peptide length set to 30
Min precursor m/z set to 360
Max precursor m/z set to 1500
Min precursor charge set to 1
Max precursor charge set to 4
Cysteine carbamidomethylation enabled as a fixed modification
Maximum number of variable modifications set to 1
Modification UniMod:35 with mass delta 15.9949 at M will be considered as variable
Modification UniMod:1 with mass delta 42.0106 at *n will be considered as variable
A spectral library will be created from the DIA runs and used to reanalyse them; .quant files will only be saved to disk during the first step
Highly heuristic protein grouping will be used, to reduce the number of protein groups obtained; this mode is recommended for benchmarking protein ID numbers; use with caution for anything else
When generating a spectral library, in silico predicted spectra will be retained if deemed more reliable than experimental ones
Fixed-width center of each elution peak will be used for quantification
Interference removal from fragment elution curves disabled
Modification UniMod:1001 with mass delta 16.0282 at K will be considered as variable
Modification UniMod:1002 with mass delta 32.0564 at K will be considered as variable
Maximum number of variable modifications set to 4
DIA-NN will use deep learning to predict spectra/RTs/IMs even for peptides carrying modifications which are not recognised by the deep learning predictor. In this scenario, if also generating a spectral library from the DIA data or using the MBR mode, it might or might not be better (depends on the data) to also use the --out-measured-rt option - it's recommended to test it with and without this option
Maximum number of missed cleavages set to 3
DIA-NN will optimise the mass accuracy automatically using the first run in the experiment. This is useful primarily for quick initial analyses, when it is not yet known which mass accuracy setting works best for a particular acquisition scheme.
The following variable modifications will be scored: UniMod:1 
Unless the spectral library specified was created by this version of DIA-NN, it's strongly recommended to specify a FASTA database and use the 'Reannotate' function to allow DIA-NN to identify peptides which can originate from the N/C terminus of the protein: otherwise site localisation might not work properly for modifications of the protein N-terminus or for modifications which do not allow enzymatic cleavage after the modified residue

11 files will be processed
[0:00] Loading spectral library H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\2nd_test_no-cut-after_mod_K\output_5th_report.predicted.speclib
[2:56] Library annotated with sequence database(s): H:\0_fasta\UniProtKB_Rat_Rev_Unrev_54764_seq_20230306.fasta
[4:08] Spectral library loaded: 54690 protein isoforms, 100304 protein groups and 108544398 precursors in 39496192 elution groups.
[4:08] Loading protein annotations from FASTA H:\0_fasta\UniProtKB_Rat_Rev_Unrev_54764_seq_20230306.fasta
[5:16] Annotating library proteins with information from the FASTA database
[5:16] Gene names missing for some isoforms
[5:16] Library contains 54690 proteins, and 24377 genes
[7:17] Initialising library

[10:34] First pass: generating a spectral library from DIA data
[10:34] File #1/11
[10:34] Loading run H:\2023-07-21-Chunli-Dimethyl-DIA runs\2023-07-22-Rat_BrainTissue_control_R1.raw
[15:04] 108452019 library precursors are potentially detectable
[15:14] Processing...
[47:40] RT window set to 8.42353
[47:40] Peak width: 3.824
[47:40] Scan window radius set to 8
[47:51] Recommended MS1 mass accuracy setting: 21.3388 ppm
[88:46] Optimised mass accuracy: 17.4178 ppm
[144:09] Removing low confidence identifications
[144:15] Searching PTM decoys
[145:28] Removing interfering precursors
[145:51] Training neural networks: 85438 targets, 92857 decoys

DIA-NN exited
DIA-NN-plotter.exe "H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\report.stats.tsv" "H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\report.tsv" "H:\2023-07-21-Chunli-Dimethyl-DIA runs\DIA-NN-test\5th_all_files\report.pdf"
PDF report will be generated in the background

**Comment by vdemichev** [comment]
The crash is likely due to the lack of memory. I would suggest to always try with mimimal modification of default settings first, and then see if changing the default settings benefits your project - the settings used here were quite extreme. --no-cut-after-mod should be used without square brackets and with full mod name, not just the number.

Best,
Vadim
